src/leafdoc.js

```diff
@@ -395,17 +402,18 @@ Leafdoc.prototype.addStr = function(str, isSource) {
 
 // 						console.log(content, ', ', alt);
 
-					var name, params = {}, type = null, defaultValue = null;
+					var name, params = {}, type = null, defaultValue = null, optional = false;
 
 					if (content) {
 						var split = regexps.functionDefinition.exec(content);
 						if (!split) {
 							console.error('Invalid ' + directive + ' definition: ', content);
 						} else {
 							name = split[1];
-							paramString = split[2];
-							type = split[3];
-							defaultValue = split[4];
+							optional = split[2] == '?';
```

Comment:
Perhaps you can provide a template which uses the `optional` property? That'd illustrate it better.